src/ext/ca/serverca/scasched/scauser.cpp

```diff
@@ -515,7 +515,7 @@ HRESULT ScaUserExecute(
         if (wzDomain && *wzDomain)
         {
             er = ::DsGetDcNameW(NULL, wzDomain, NULL, NULL, NULL, &pDomainControllerInfo);
-            if (HRESULT_FROM_WIN32(er) == RPC_S_SERVER_UNAVAILABLE)
+            if (RPC_S_SERVER_UNAVAILABLE == er)
```

Comment:
No, I edited that area because er and hr dragged on their error values to the return value.
However, MSDN does state that retrying the function with DS_FORCE_REDISCOVERY is the thing to do when the cached domain is not available.